### DISCUSSION TITLE
Can I customize the default layout

### DISCUSSION THREAD

**Discussion by trry-hub** [vidstack/player#1351] [General]
I want to modify the thumbnail component. I need to input HTML. How can I achieve this?
environment
vue3 + vite

**Comment by mihar-22** [comment]
For Web Components, you can customize using [CSS variables](https://www.vidstack.io/docs/wc/player/components/layouts/default-layout?styling=default-theme#css-variables) or any of the options listed on the linked page. You can also just use CSS to target and customize whatever you want. Inspect your document like you normally would and apply styles using the classes. For example, the thumbnails have the `vds-thumbnail` class.

**Reply by trry-hub** [reply]
But in the thumbnail section, the HTML structure is not what I want, and I need to modify the content in the tooltip. The content I want to showcase is not just a title. So I think if the default layout can be designed with each part as a slot, it may be more convenient to use

**Reply by mihar-22** [reply]
Slots are not available for Web Components at the moment due to some limitations with Shadow DOM.

**Reply by trry-hub** [reply]
Is there any other way? I need to use the following style in my business, currently implemented using ArtPlayer's custom plugins. But I really want to use Vidstack, and I believe it has better underlying design than Artplayer.

![image](https://github.com/user-attachments/assets/fb08d156-e098-4f3b-80d8-257393c4d151)

**Reply by trry-hub** [reply]
Can you update the demo for the Vue version? Currently, it cannot achieve the layout effect

**Reply by mihar-22** [reply]
The alternative is to build custom layout, see our [examples](https://github.com/vidstack/examples/tree/main/player) and the docs for full customization options and styling for each component. You can use our Default Theme so you don't have to style things from scratch and get the same design as our Default Layout for buttons, sliders, menus, etc.

**Reply by trry-hub** [reply]
Completely customizing is too difficult for front-end development engineers without UI design experience. And without reading the Vidstack source code, I am not proficient in using components
